src/zenml/config/secrets_store_config.py

```diff
@@ -31,9 +33,51 @@ class SecretsStoreConfiguration(BaseModel):
 
     Attributes:
         type: The type of store backend.
+        class_path: The Python class path of the store backend. Should point to
+            a subclass of `BaseSecretsStore`. This is optional and only
+            required if the store backend is not one of the built-in
+            implementations.
     """
 
     type: SecretsStoreType
+    class_path: Optional[str] = None
+
+    @root_validator
+    def validate_external(cls, values: Dict[str, Any]) -> Dict[str, Any]:
+        """Validate that class_path is set for external secrets stores.
+
+        Args:
+            values: Dict representing user-specified runtime settings.
+
+        Returns:
+            Validated settings.
+
+        Raises:
+            ValueError: If class_path is not set when using an external secrets
+                store.
+        """
+        if not values.get("type"):
+            return values
+        if values["type"] == SecretsStoreType.CUSTOM:
+            if values["class_path"] is None:
+                raise ValueError(
+                    "A class_path must be set when using a custom secrets "
+                    "store implementation."
+                )
+        elif (
+            values["type"]
+            in [
+                SecretsStoreType.SQL,
+                SecretsStoreType.REST,
```

Comment:
Is the class path allowed for aws/gcp/etc?

Reply:
This is a left-over from another iteration during which I kept all secrets store implementations inside integrations, but now this can go away.